plugins/lsp/language-servers/_mk-lsp.nix

```diff
@@ -50,16 +50,32 @@ in
       package =
         if lib.isOption package then
           package
-        else
+        else if args ? package then
           lib.mkPackageOption pkgs name {
             nullable = true;
             default = package;
+          }
+        else
+          # If we're not provided a package, we should provide a no-default option
+          lib.mkOption {
+            type = types.nullOr types.package;
+            description = ''
+              The package to use for ${name}. Has no default, but can be set to null.
+            '';
           };
 
       cmd = mkOption {
         type = with types; nullOr (listOf str);
         default =
-          if (cfg.package or null) != null then if builtins.isFunction cmd then cmd cfg else cmd else null;
+          # TODO: do we really only want the default `cmd` when `package` is non-null?
+          if !(opt.package.isDefined or false) then
+            null
+          else if cfg.package == null then
+            null
+          else if builtins.isFunction cmd then
+            cmd cfg
+          else
+            cmd;
```

Comment:
I don't want to address this in this PR, but I've added a TODO comment regardless.

Essentially, `plugins.lsp.servers.<name>.cmd` is eventually passed through to `plugins.lsp.enabledServers.*.extraOptions`, so in a way it is a freeform sub-option.

It also seems that whether `cfg.package` is null or not is unrelated to whether we should use the default cmd.

Perhaps this comes from our history of defining a command based on things like `lib.getExe cfg.package`? Hopefully we've moved away from doing that and can enforce a policy where default `cmd` values do not reference the nix-store and are always defined.

If we keep the current behavior, we should also have a `defaultText` that explains how the default is conditional.